Add notice subtype (#2909)
nasa-gcn/gcn.nasa.gov: app/routes/docs.notices.schema.mdx

Mapping of custom, mission-specific fields. The first property, `'$schema: { type: "string", format: "uri" }'`, must be included when using `'unevaluatedProperties': false`, otherwise the inclusion of the `$schema` property in your example files will cause validation to fail. Field names should use [snake_case](https://en.wikipedia.org/wiki/Snake_case). Each entry should at least have a description and a type. See the [list of available data types](https://json-schema.org/understanding-json-schema/reference/type.html).

> We suggest including a `subtype` property for situations where a single schema is used to produce multiple types of notices. This property should be customizable to fit the specific mission and should be designed as an enumerated list. For example, in space missions, the enum might include values such as ground and flight for distinct notice types.

Reviewer: We're still calling it "types of notices", when we mean there's some distinguishing criteria.  Maybe instead call it "variations of this notice".

Author: ```suggestion
We suggest including a `subtype` property for situations where a single schema is used to produce multiple variations of notices. This property should be customizable to fit the specific mission and should be designed as an enumerated list. For example, in space missions, the enum might include values such as ground and flight for distinct notice types.
```

How about "multiple subtypes of notices"?

```json
"subtype": {
"enum": ["type_1", "type_2", "type_3"],

> "description": "Defines the subtype of notice. It is recommended when a single schema is used to produce multiple types of notices."

Author: ```suggestion
"description": "Defines the subtype of notice. It is recommended when a single schema is used to produce multiple variations of a notice."
```

Mapping of custom, mission-specific fields. The first property, `'$schema: { type: "string", format: "uri" }'`, must be included when using `'unevaluatedProperties': false`, otherwise the inclusion of the `$schema` property in your example files will cause validation to fail. Field names should use [snake_case](https://en.wikipedia.org/wiki/Snake_case). Each entry should at least have a description and a type. See the [list of available data types](https://json-schema.org/understanding-json-schema/reference/type.html).

> We recommend designing the new Notices schema with a minimal number of Kafka topics, preferably one schema per instrument; however, exceptions may be made when necessary. For cases where a single schema produces multiple types of notices, please include a `notice_type` field defined as an enumeration and design the enum list according to your requirements. For example, in space missions, the enum might include values such as ground and flight for distinct notice types.

Reviewer: ```suggestion
We recommend designing the new Notices schema with a minimal number of Kafka topics, preferably one schema per instrument; however, exceptions may be made when necessary. For cases where a single schema produces multiple types of notices, please include a `notice_type` field defined as an enumeration and design the enum list according to your requirements, and an example of each usage if parameters differ. For example, in space missions, the enum might include values such as ground and flight for distinct notice types.
```

> 

Author: We don't have any example yet. 
Only upcoming new Notice, Fermi/GBM utilizing it.

> 

Reviewer: add code block of notice type example, rather than just in word in the paragraph

> 

Author: Done

Mapping of custom, mission-specific fields. The first property, `'$schema: { type: "string", format: "uri" }'`, must be included when using `'unevaluatedProperties': false`, otherwise the inclusion of the `$schema` property in your example files will cause validation to fail. Field names should use [snake_case](https://en.wikipedia.org/wiki/Snake_case). Each entry should at least have a description and a type. See the [list of available data types](https://json-schema.org/understanding-json-schema/reference/type.html).

We suggest including a `subtype` property for situations where a single schema is used to produce multiple types of notices. This property should be customizable to fit the specific mission and should be designed as an enumerated list. For example, in space missions, the enum might include values such as ground and flight for distinct notice types.

```json
"subtype": {
"enum": ["type_1", "type_2", "type_3"],

> "description": "Defines the subtype of notice. It is recommended when a single schema is used to produce multiple type of notices."

Author: ```suggestion
"description": "Defines the subtype of notice. It is recommended when a single schema is used to produce multiple types of notices."
```

}
```

## Extended Example

> We encourage exploring our [schema browser](/docs/schema/stable/gcn/notices) and [GitHub repository](https://github.com/nasa-gcn/gcn-schema/tree/main/gcn/notices) for more examples. Here is an extended example from Swift/BAT-GUANO:

Author: ```suggestion
We recommend exploring our [schema browser](/docs/schema/stable/gcn/notices) and [GitHub repository](https://github.com/nasa-gcn/gcn-schema/tree/main/gcn/notices) for more examples. Here is an extended example from Swift/BAT-GUANO:
```
